pretyping/unification.ml

```diff
@@ -191,7 +191,8 @@ let pose_all_metas_as_evars env evd t =
   let rec aux t = match EConstr.kind !evdref t with
   | Meta mv ->
       (match Evd.meta_opt_fvalue !evdref mv with
-       | Some ({rebus=c},_) -> c
+       | Some ({rebus=c;freemetas=mvs},_) ->
+         if Evd.Metaset.is_empty mvs then c else aux c
```

Comment:
Should we `meta_reassign` here?

Reply:
Yes perhaps. I´ll try tomorrow if that will fix the metacoq issue.